### ISSUE TITLE
Add firestore support

### ISSUE THREAD

**Issue by hmbenhaim** [appsup-dart/firebase_dart#21]
Hi thank you for your package, I just wander if you going to firestore support in this package any time soon,
Thank you.

**Comment by worldwidee** [comment]
yes when it supports firestore it will be a perfect package <3

**Comment by rbellens** [comment]
Thanks. I'd like to add support for firestore, but as this is a hobby project, I don't currently have the time to do so. However, if you really need it, you could consider sponsorship. That would allow me to dedicate more time to this project.

**Comment by worldwidee** [comment]
I wish i could do that. I couldn't earn money yet. You helped a lot to who already have a mobile application using firebase and want to make a desktop version . I am devoloping the app for all platforms. And thank to you, i am using firebase storage in desktop version. Firebase should have develope desktop version long before. Luckily flutterfire decided to develope desktop version but its extremly slow.

**Comment by hotdang-ca** [comment]
@rbellens I would love to help implement this by contributing, though I would need some pointers on getting started.

**Comment by rbellens** [comment]
Hi @hotdang-ca , would be great. 

What info would you need?

**Comment by hotdang-ca** [comment]
Maybe just an annotated walk-through on how the bootstrap works (initialization --> ready).

oh and, free time :D free time would be great!

**Comment by rbellens** [comment]
hi @hotdang-ca 

I've created a dummy implementation based on the flutterfire api interface (in branch `feature/firestore`).

It has many missing implementations, but getting a document from the server (without authentication) works already. See the example in the example directory.

I guess, you can start filling in many of the missing implementations quite easily. To implement the more advanced features (e.g. caching), you'll probably need to base the implementation on the official android or js repos.

**Comment by MiniSuperDev** [comment]
hello, is there any progress in the implementation of firestore?

**Comment by WH-Creations** [comment]
Is this project still actively supported/alive?

**Comment by rbellens** [comment]
yes it is, but support for firestore is currently not in the pipeline. I do welcome pull requests. You could have a look at the `feature/firestore` branch which contains a very basic implementation for firestore. 
You could also consider sponsorship. That would allow me to spend more time on this project and give this feature priority.
